### PR TITLE
Feature/create UI popup call

### PR DESCRIPTION
**Pull request by PhamNhuQuocBao**
# Description

## Detailed PR Verification Procedure(s)
none.

## Ticket

[GS-46](https://tito-training.atlassian.net/jira/software/projects/GS/boards/1?selectedIssue=GS-46)

## Application Checklist

- [x] **JIRA Linked**: Author has followed template and has a title in the following format:
  > JIRA-0: Generic summary of work done in PR
- [x] **No Breaking Changes**: Author has confirmed all API contract changes are non-breaking.
- [ ] **Changes Unit Tested**: Author has unit tested new code to prove it works.
- [ ] **Application Passes All Tests**: Author ensures this pull-request passes `npm test`.
- [x] **Author Self Reviewed**: Author of pull-request has self-reviewed this code.
- [x] **Author is Assignee**: Author has assigned self as assignee of pull request.

## Security Checklist

- [ ] **Code Review for Security:** Developer has self-reviewed this code, based on OWASP Top-10 vulnerabilities and general environment security attributes, to avoid dversely impact the security of the system/environment/application, to the best of their abilities.
- [ ] **Backout Procedures:** Developer has considered how the changes presented in this pull request can be replaced with a prior version of these changes, fixed going forward, and/or implemented to be backwards compatible.

### CODE REVIEW

**Comment by duong56** [comment]
please split file changes

File: src/components/common/PopupCall/components/Content.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+interface Props {
+  type: 'call to' | 'call away'
+  name: string
+  avatar?: string
+  phoneNumber: string
+}
+
+const Content: React.FC<Props> = ({ type, name, avatar, phoneNumber }) => {
+  const call = type === 'call to' ? callTo : callAway
```
Comment (duong56) [inline src/components/common/PopupCall/components/Content.tsx:14]: Please explain this one

File: src/components/common/PopupCall/components/Content.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  return (
+    <>
+      <div className='content__top'>
+        <picture className='avatar'>
+          <img src={avatar} alt='' />
+        </picture>
+        <figure className='call--type'>
```
Comment (duong56) [inline src/components/common/PopupCall/components/Content.tsx:22]: why do you use picture and figure tag at the same time

File: src/components/common/PopupCall/components/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import Connected from './StateCall/Connected'
+import Disconnect from './StateCall/Disconnect'
+import Ring from './StateCall/Ring'
+
+interface Props {
+  state: string
+  type: 'call to' | 'call away'
+  onOK?: () => void
+  onCancel?: () => void
+  timing?: string | undefined
```
Comment (duong56) [inline src/components/common/PopupCall/components/Footer.tsx:11]: undefined unnecessary

File: src/components/common/PopupCall/components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+const Footer: React.FC<Props> = ({
+  type,
+  state,
+  onCancel,
+  onOK,
+  timing,
+  isConnected,
+}) => {
+  switch (state) {
+    case 'ring':
```
Comment (duong56) [inline src/components/common/PopupCall/components/Footer.tsx:24]: please define constant

File: src/components/common/PopupCall/components/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      return <Ring type={type} onCancel={onCancel} onOK={onOK} />
+
+    case 'connected':
```
Comment (duong56) [inline src/components/common/PopupCall/components/Footer.tsx:27]: same

File: src/components/common/PopupCall/components/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      return (
+        <>
+          {isConnected && <p className='time'>{timing}</p>}
```
Comment (duong56) [inline src/components/common/PopupCall/components/Footer.tsx:30]: duplicate code here

File: src/components/common/PopupCall/components/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React, { memo } from 'react'
+
+interface Props {
+  type: 'call to' | 'call away'
+  phoneNumber: string
+}
+
+const Header: React.FC<Props> = ({ type, phoneNumber }) => {
+  const day = new Date()
+  const timeStart = `${day.getHours()}:${day.getMinutes()}`
```
Comment (duong56) [inline src/components/common/PopupCall/components/Header.tsx:10]: please search about format date string in this case.

File: src/components/common/PopupCall/components/StateCall/Connected.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+interface Props {
+  onCancel?: () => void
+}
+
+const Connected: React.FC<Props> = ({ onCancel }) => {
+  return (
+    <>
+      <p className='status'>Đã kết nối</p>
+      <button onClick={onCancel}>
+        <figure>
```
Comment (duong56) [inline src/components/common/PopupCall/components/StateCall/Connected.tsx:13]: unnecessary

File: src/components/common/PopupCall/components/StateCall/Disconnect.tsx [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import React, { memo } from 'react'
+
+const Disconnect: React.FC = () => {
+  return (
+    <>
```
Comment (duong56) [inline src/components/common/PopupCall/components/StateCall/Disconnect.tsx:5]: unnecessary

File: src/components/common/PopupCall/components/StateCall/Disconnect.tsx [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import React, { memo } from 'react'
+
+const Disconnect: React.FC = () => {
```
Comment (duong56) [inline src/components/common/PopupCall/components/StateCall/Disconnect.tsx:3]: unnecessary

File: src/components/common/PopupCall/components/StateCall/Ring.tsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+interface Props {
+  type: 'call to' | 'call away'
+  onOK?: () => void
+  onCancel?: () => void
+}
+
+const Ring: React.FC<Props> = ({ type, onCancel, onOK }) => {
+  switch (type) {
+    case 'call to':
```
Comment (duong56) [inline src/components/common/PopupCall/components/StateCall/Ring.tsx:13]: please define constants

**Comment by duong56** [comment]
Please check comment

File: src/components/common/PopupCall/components/Content.tsx (hunk 1 above)
Comment (PhamNhuQuocBao) [reply src/components/common/PopupCall/components/Content.tsx:14]: cái này là xét xem type là gì. Nếu type là call to thì sẽ gán icon callTo cho biến call, sau đó em đưa biến call vào att src của thẻ img.

**Pull request closed without merging**